src/Domain/EmbeddingMaps/ProductMaps.hpp

```diff
@@ -0,0 +1,118 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines the class templates ProductOf2Maps and ProductOf3Maps.
+
+#pragma once
+
+#include <memory>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "EmbeddingMap.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Parallel/PupStlCpp11.hpp"
+
+namespace EmbeddingMaps {
+
+/// Product of two codimension=0 EmbeddingMaps.
```

Comment:
Add to `ComputationalDomain` group. Should we have a group for embedding maps maybe? I think that's the right thing to do if `ComputationalDomain` gets a lot of things. But since AMR will be separate, I think it would be fine to have them in `ComputationalDomain` and not have an `EmbeddingMaps` group. The nampsace should be enough